src/NServiceBus.Core/Transports/Sending.cs

```diff
@@ -1,53 +1,60 @@
+#pragma warning disable 1591
 namespace NServiceBus
 {
-    using System;
-    using System.Threading;
-    using System.Threading.Tasks;
-    using Features;
+    using ObjectBuilder;
+    using Routing;
+    using Settings;
     using Transport;
 
-    class Sending : Feature
+    public class TransportComponent
     {
-        public Sending()
+        public TransportComponent(TransportInfrastructure transportInfrastructure)
         {
-            EnableByDefault();
+            TransportInfrastructure = transportInfrastructure;
         }
 
-        protected internal override void Setup(FeatureConfigurationContext context)
-        {
-            var transport = context.Settings.Get<OutboundTransport>();
-            var lazySendingConfigResult = new Lazy<TransportSendInfrastructure>(() => transport.Configure(context.Settings), LazyThreadSafetyMode.ExecutionAndPublication);
-            context.Container.ConfigureComponent(c =>
-            {
-                var dispatcher = lazySendingConfigResult.Value.DispatcherFactory();
-                return dispatcher;
-            }, DependencyLifecycle.SingleInstance);
+        public IDispatchMessages Dispatcher { get; set; }
```

Comment:
@timbussmann At some point I proposed using something like `FutureComponent<T>` for things like this one but my proposal got rejected (I don't remember the ID of the PR). The advantage was that when you tried to resolve the instance of `T` before the start, it would throw. The disadvantage was that you had to pass it to the class that used it. 

We could use a built-in `Lazy<T>` for the same thing, passing a `func` that ensures we are in appropriate stage to get value but I don't like it either. 

Since 99% of the time the dispatcher will be used correctly (after starting) I think the solution (if any) should be friction-less for that success path. My idea was something like this:

```
class DispatcherWrapper : IDispatchMessages
{
   IDispatchMessages realDispatcher //passed in ctor
   Func<...,Task> dispatcherInvocationCallback = (args) => { throw new Exception("Boom!") };

   public void Enable()
   {
      dispatcherInvocationCallback = (args) => realDispatcher.Dispatch(args);
   } 
  
   public Task Dispatch(args)
   {
      return dispatcherInvocationCallback(args);
   }
}
```

This way if would look and act like a normal dispatcher unless somebody tries to play with it before it is enabled.